### PR TITLE
Brace highlight

### PR DESCRIPTION
**Pull request by HuanhuanSunMSFT**
@AndreSayreMSFT @EricMSFT @Microsoft/poshtools

### CODE REVIEW

File: PowerShellTools/Classification/PowerShellBraceMatchingTagger.cs [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+namespace PowerShellTools.Classification
+{
+    internal sealed class PowerShellBraceMatchingTagger : ITagger<TextMarkerTag>, INotifyBufferUpdated
+    {
+	ITextView _textView;
+	ITextBuffer _textBuffer;
+	SnapshotPoint? _caretPos;
+	private IDictionary<int, int> _startBraces;
+	private IDictionary<int, int> _endBraces;
+	private const string MarkedColor = "blue";
```
Comment (ASayre) [inline PowerShellTools/Classification/PowerShellBraceMatchingTagger.cs:17]: Does this need to be themed?  How about on high-contrast themes?

File: PowerShellTools/Classification/PowerShellTokenizationService.cs [hunk 2]
```diff
@@ -88,6 +88,7 @@ private void UpdateTokenization()
                             NotifyOnTagsChanged(BufferProperties.Classifier, currentSnapshot);
                             NotifyOnTagsChanged(BufferProperties.ErrorTagger, currentSnapshot);
                             NotifyOnTagsChanged(typeof(PowerShellOutliningTagger).Name, currentSnapshot);
+			    NotifyBufferUpdated();
```
Comment (ASayre) [inline PowerShellTools/Classification/PowerShellTokenizationService.cs:91]: nit: spacing.

File: PowerShellTools/Classification/PowerShellTokenizationService.cs [hunk 3]
```diff
@@ -109,6 +110,16 @@ private void NotifyOnTagsChanged(string name, ITextSnapshot currentSnapshot)
...
         }
 
+	private void NotifyBufferUpdated()
+	{
+	    INotifyBufferUpdated tagger;
+	    if (_textBuffer.Properties.TryGetProperty<INotifyBufferUpdated>(typeof(PowerShellBraceMatchingTagger).Name, out tagger) && tagger != null)
+	    {
+		tagger.OnBufferUpdated(_textBuffer);
+	    }
+
```
Comment (ASayre) [inline PowerShellTools/Classification/PowerShellTokenizationService.cs]: nit: spacing

**Comment by ASayre** [comment]
Confirm: No perf impact on large files? :)

File: PowerShellTools/Classification/PowerShellTokenizationService.cs (hunk 3 above)
Comment (HuanhuanSunMSFT) [inline PowerShellTools/Classification/PowerShellTokenizationService.cs]: Fixed.

File: PowerShellTools/Classification/PowerShellTokenizationService.cs (hunk 2 above)
Comment (HuanhuanSunMSFT) [inline PowerShellTools/Classification/PowerShellTokenizationService.cs:91]: This looks good in VS. Looks like Github usually messes up the indentation and spacing.

File: PowerShellTools/Classification/PowerShellBraceMatchingTagger.cs (hunk 1 above)
Comment (HuanhuanSunMSFT) [inline PowerShellTools/Classification/PowerShellBraceMatchingTagger.cs:17]: Good catch. I tried this on high-contrast themes, not obvious. May need your help on how turn this into different colors based on theme mode.

**Comment by HuanhuanSunMSFT** [comment]
No perf impact. I tried on the one with 22 thousand lines and it just works smoothly. As you may find how we find the paired braces, we either get it from the ones already parsed or we search it line by line. I mean it's quite efficient. @AndreSayreMSFT

**Comment by ASayre** [comment]
Looks good once AppVeyor's happy!

**Pull request merged**